runtime/services/catalog/artifacts/yaml/objects.go

```diff
@@ -16,14 +16,27 @@ import (
  */
 type Source struct {
 	Type                  string
-	Path                  string `yaml:"path,omitempty"`
-	CsvDelimiter          string `yaml:"csv.delimiter,omitempty" mapstructure:"csv.delimiter,omitempty"`
-	URI                   string `yaml:"uri,omitempty"`
-	Region                string `yaml:"region,omitempty" mapstructure:"region,omitempty"`
-	GlobMaxTotalSize      int64  `yaml:"glob.max_total_size,omitempty" mapstructure:"glob.max_total_size,omitempty"`
-	GlobMaxObjectsMatched int    `yaml:"glob.max_objects_matched,omitempty" mapstructure:"glob.max_objects_matched,omitempty"`
-	GlobMaxObjectsListed  int64  `yaml:"glob.max_objects_listed,omitempty" mapstructure:"glob.max_objects_listed,omitempty"`
-	GlobPageSize          int    `yaml:"glob.page_size,omitempty" mapstructure:"glob.page_size,omitempty"`
+	Path                  string         `yaml:"path,omitempty"`
+	CsvDelimiter          string         `yaml:"csv.delimiter,omitempty" mapstructure:"csv.delimiter,omitempty"`
+	URI                   string         `yaml:"uri,omitempty"`
+	Region                string         `yaml:"region,omitempty" mapstructure:"region,omitempty"`
+	GlobMaxTotalSize      int64          `yaml:"glob.max_total_size,omitempty" mapstructure:"glob.max_total_size,omitempty"`
+	GlobMaxObjectsMatched int            `yaml:"glob.max_objects_matched,omitempty" mapstructure:"glob.max_objects_matched,omitempty"`
+	GlobMaxObjectsListed  int64          `yaml:"glob.max_objects_listed,omitempty" mapstructure:"glob.max_objects_listed,omitempty"`
+	GlobPageSize          int            `yaml:"glob.page_size,omitempty" mapstructure:"glob.page_size,omitempty"`
+	HivePartition         *bool          `yaml:"hive_partitioning,omitempty" mapstructure:"hive_partitioning,omitempty"`
+	Timeout               int            `yaml:"timeout,omitempty" mapstructure:"timeout,omitempty"`
+	Policy                *ExtractPolicy `yaml:"extract,omitempty" mapstructure:"source.extract,omitempty"`
```

Comment:
1. I thought its idiomatic in golang to remove type context from variable name.
2. mapstructure is actually not required here. Removed.

Reply:
Yeah, it's not clear cut. When things have a separate name, then you're right, like `TimeoutDuration time.Duration` would be `Timeout time.Duration`. But when the name and type name happen to match, you just use the type name (for example, `Duration time.Duration`).

However, since this represents the YAML, it actually might make sense to just mirror the yaml and call it `Extract`?